cfme/tests/automate/custom_button/test_service_objects.py

```diff
@@ -492,7 +526,46 @@ def test_custom_button_role_access_service(context):
             4. Create custom button with role
             5. Check use able to access custom button or not
     """
-    pass
+    usr, role = user_self_service_role
+    service = MyService(appliance, name=gen_rest_service.name)
+
+    # custom button on group with role
+    with appliance.context.use(ViaUI):
+        btn = service_button_group.buttons.create(
+            text=fauxfactory.gen_alphanumeric(start="btn_"),
+            hover=fauxfactory.gen_alphanumeric(start="hvr_"),
+            system="Request",
+            request="InspectMe",
+            roles=[role.name],
+        )
+        request.addfinalizer(btn.delete_if_exists)
+
+    # check button with admin and other user for UI and SSUI
+    for user in [usr, appliance.user]:
+        with user:
+            with appliance.context.use(context):
+                logged_in_page = appliance.server.login(user)
+                navigate_to = ssui_nav if context is ViaSSUI else ui_nav
```

Comment:
Consider setting `navigate_to` and `cb_group_class` in an if block, since both depend on the context.

```
if context is ViaSSUI:
    navigate_to = ssui_nav
    group_class = CustomButtonSSUIDropdown
else:
    navigate_to = ui_nav
    group_class = Dropdown

view = navigate_to(service, 'Details')
cb_group = group_class(view, service_button_group.text)
```